sanitize.go

```diff
@@ -83,17 +84,23 @@ func (p *Policy) SanitizeBytes(b []byte) []byte {
 // It returns a bytes.Buffer containing the HTML that has been sanitized by the
 // policy. Errors during sanitization will merely return an empty result.
 func (p *Policy) SanitizeReader(r io.Reader) *bytes.Buffer {
-	return p.sanitize(r)
+	return p.sanitizeWithBuff(r)
+}
+
+// SanitizeReaderWriter takes an io.Reader that contains a HTML fragment or document
+// and applies the given policy whitelist and writes to the provided writer returning
+// an error if there is one.
+func (p *Policy) SanitizeReaderToWriter(r io.Reader, w io.Writer) error {
```

Comment:
Why do you need the error returned here?

The original design was to be safe by default and to sanitize in an unattended way... I couldn't see anything useful that I could do with an error... that is I cannot clean it as build clean HTML is a hard task (and subjective depending on what the HTML renderer is capable of), and so garbage in = garbage out was the design philosophy. The only thing we'd guarantee is that the output should be safe from an XSS perspective to be rendered.

Do you have a strong case for requiring an error to be returned?

Reply:
It's possible that the reader or the writer returns an error on the write/read and the upstream caller will want to use it.

For example, say we have a writer in the stream that will only allow say 10mb to be written - then that writer will return an error if we write more than 10mb and it will be helpful for the caller to know that.